### DISCUSSION TITLE
Is there any difference when we pack out project with webpack by yarn or pnpm?

### DISCUSSION THREAD

**Discussion by lanmch** [webpack/webpack#14932] [Q&A]
In my project, there is some rules to translate the node modules, like this translate svg to react component
```
{
    test: /\.svg$/,
    include: [ project.paths.inline_icons() ], // the real path is '/Users/mc/project/no-commit/bear/node_modules/@ud/icons/icons'
    use: reactSvgLoader,
}
```
But when I run my project, I find it not be translated.
I am not sure if it is caused by the npm tool source（Although I don’t think the tool source will cause this reason）.
But in fact, because of pnpm, the real path what code use is '/Users/mc/project/no-commit/bear/node_modules/.pnpm/@ud+icons@1.43.0/node_modules/@ud/icons/icons'.

Is there any one hav the same problem?

**Comment by alexander-akait** [comment]
What is directory where your store modules?

**Reply by lanmch** [reply]
store modules? means what

**Reply by alexander-akait** [reply]
Oh, I missed you store them in `node_modules`, there are only one different - https://github.com/webpack/webpack/blob/main/lib/config/defaults.js#L361

**Reply by alexander-akait** [reply]
Look at code there we use `process.versions.pnp`, but I think you problem is not related to webpack or code above, just use valid `include`, you can use `require.resolve` to get package location

**Reply by lanmch** [reply]
Oh, i think we just talk about something different hah~
Let me show you the detail.
From the below config, I try to convert some svg to react component by webpack. Now in my project, I try to import the svg(react component)，but it is not converted. 
when i use the yarn, it works well. But when I use the pnpm, it not works.
I think maybe it is because the pnpm construct? because when we import in yarn, the real path is bear/node_modules/@ud/icons/icons. But with pnpm, the real path is bear/node_modules/.pnpm/@ud@1.2.1/node_modules/@ud/icons/icons. So the webpack pack path is not same as the real path

**Reply by alexander-akait** [reply]
webpack is agnostic to package managers, yes we have a little code for yarn PnP and no more

**Reply by alexander-akait** [reply]
If you have another path, you need to fix your `test`/`include`/`exclude`

**Reply by lanmch** [reply]
Yes, I think webpack is agnostic to package managers. The same configuration should be used for yarn/npm/pnpm rightly. But my problem occur when change to pnpm. So I want to know if you have hear the same problem or hav some advice.

**Reply by lanmch** [reply]
if I include the path ‘'/Users/mc/project/no-commit/bear/node_modules/.pnpm/@ud+icons@1.43.0/node_modules/@ud/icons/icons'’, it works. But include this path looks like not universal

**Reply by alexander-akait** [reply]
What is return `require.resolve("@ud/icons")`?

**Reply by lanmch** [reply]
I'll try it later~ thx
